### PR TITLE
Migrate S5773: Scaffold the rule for the new SE

### PR DESCRIPTION
**Pull request by Tim-Pohlmann**
Step 1 of #7564

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/RestrictDeserializedTypesTest.cs [hunk 1]
```diff
@@ -18,69 +18,116 @@
...
+            .Verify();
+    }
+
+    [Ignore] // ToDo: Remove after implementation
+    [TestMethod]
+    public void RestrictDeserializedTypesFormatters_Roslyn_CSharp8()
+    {
+        using var _ = new AssertIgnoreScope(); // EnsureStackState fails an assertion in this test file
+        roslynCS.AddPaths("RestrictDeserializedTypes.cs")
+            .WithOptions(ParseOptionsHelper.FromCSharp8)
```
Comment (zsolt-kolbay-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/RestrictDeserializedTypesTest.cs:60]: Split it into two test files:
- **RestrictDeserializedTypes.cs**: test cases which need only C# 7.3
- **RestrictDeserializedTypes.CSharp8.cs**: test cases which use C# 8.0 language features (like coalescing assignment)

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/RestrictDeserializedTypes.cs
[the remark this answers is not in the document]
Comment (Tim-Pohlmann) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/RestrictDeserializedTypes.cs:1]: "safe" options are covered by `UnknownFormattersAreSafeByDefault`.

**Review by zsolt-kolbay-sonarsource — APPROVED** [review]
LGTM

**Comment by Tim-Pohlmann** [comment]
@pavel-mikula-sonarsource please merge this scaffolding PR.

**Pull request merged**